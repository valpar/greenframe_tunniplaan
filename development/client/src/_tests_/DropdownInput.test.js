import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DropdownInput from '../components/UI/Dropdown/DropdownInput.js';


describe('DropdownInput', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<DropdownInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders label if provided", () => {
    render(<DropdownInput label="Test Label" index={0} />);
    expect(screen.getByText('Test Label')).toBeInTheDocument();
  });
  it("shows tooltip when mouse enters and hides when mouse leaves", async () => {
    const errorMessage = "This is an error!";
    render(<DropdownInput onErrorMessage={errorMessage} />);

    const input = screen.getByRole('textbox');
    fireEvent.mouseEnter(input);

    expect(screen.getByText(errorMessage)).toBeInTheDocument();
    fireEvent.mouseLeave(input);

    // Wait for potential effects or transitions.
    await new Promise(r => setTimeout(r, 500));
    expect(screen.queryByText(errorMessage)).not.toBeInTheDocument();
  });
});
