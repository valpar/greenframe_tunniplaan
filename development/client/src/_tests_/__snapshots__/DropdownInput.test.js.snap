// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownInput matches snapshot 1`] = `
<div
  className="relative w-full flex flex-col"
>
  <input
    autoComplete="off"
    className="p-2 border w-full border-borderGray  rounded-none font-normal focus:outline-none focus:border-black"
    name=""
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    readOnly={false}
    type="text"
    value=""
  />
</div>
`;
